module/zfs/zfs_ioctl.c

```diff
@@ -3737,7 +3738,9 @@ zfs_check_settable(const char *dsname, nvpair_t *pair, cred_t *cr)
 			    SPA_VERSION_ZLE_COMPRESSION))
 				return (SET_ERROR(ENOTSUP));
 
-			if (intval == ZIO_COMPRESS_LZ4) {
+			if (intval == ZIO_COMPRESS_LZ4 ||
+			    (intval >= ZIO_COMPRESS_LZ4HC_1 &&
```

Comment:
No new on-disk feature?  That seems wrong (you're adding new BP_GET_COMPRESS() values that other software will barf on).  But if existing lz4 decompression code can deal with blocks compressed by lz4hc, then you can remove the new BP_GET_COMPRESS() values and just use the existing LZ4 value.